### PR TITLE
Add msmtp and msmtp-sendmail

### PR DESCRIPTION
**Pull request by blmaier**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by blmaier** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/msmtp/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        - automake
+        - make
+        - pkg-config
+        - gettext
+        - texinfo
+      host:
+        - gnutls
+        - libgsasl
+      run:
+        - libgsasl
```
Comment (chrisburr) [inline recipes/msmtp/recipe.yaml:39]:
```suggestion
        - {{ pin_compatible('libgsasl', max_pin='x') }}
```

**Review by chrisburr — COMMENTED** [review]
That's a fun coninsidence (#28610). To update to 1.8.27 we need to update to libgsasl 2.x in conda-forge.

**Comment by blmaier** [comment]
> That's a fun coninsidence (#28610). To update to 1.8.27 we need to update to libgsasl 2.x in conda-forge.

What are the odds :smiley:. Would you like to be added to `recipe-maintainers`?

Yes, 1.8.26+ requires features from libgsasl 2.x. Porting msmtp 1.8.25 seemed the simplest start.

**Comment by chrisburr** [comment]
> Would you like to be added to recipe-maintainers?

I suspect my use for msmtp might be quite short term so I'll pass given someone else is interested 😄


> Yes, 1.8.26+ requires features from libgsasl 2.x. Porting msmtp 1.8.25 seemed the simplest start.

I agree. Hopefully I'll have an update PR for libgsasl today so we don't end up stuck on 1.8.25 forever :)

**Pull request merged**